src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -218,8 +219,32 @@ public static JpegFileMarker FindNextFileMarker(byte[] marker, DoubleBufferedStr
         /// <typeparam name="TPixel">The pixel format.</typeparam>
         /// <param name="stream">The stream, where the image should be.</param>
         /// <returns>The decoded image.</returns>
-        public Image<TPixel> Decode<TPixel>(Stream stream)
+        public async Task<Image<TPixel>> DecodeAsync<TPixel>(Stream stream)
             where TPixel : unmanaged, IPixel<TPixel>
+        {
+            if (stream.CanSeek)
+            {
+                return this.Decode<TPixel>(stream);
```

Comment:
I'm not sure whether this execution path is [Stephen Toub proof](https://devblogs.microsoft.com/pfxteam/should-i-expose-asynchronous-wrappers-for-synchronous-methods/), however I can live with it :smile:

I think we got better in total.